src/python/grapl_analyzerlib/grapl_analyzerlib/schema.py

```diff
@@ -0,0 +1,93 @@
+import abc
+import types
+
+from copy import deepcopy
+
+from typing import cast, Type, TypeVar, Any, Dict, Tuple, Union
+
+
+V = TypeVar("V", bound="Viewable")
+
+
+def default_properties() -> Dict[str, "PropType"]:
+    return {
+        "uid": PropType(PropPrimitive.Str, False),
+        "dgraph.type": PropType(PropPrimitive.Str, True),
+    }
+
+
+class SingletonMeta(type):
+    _instances = {}
+
+    def __call__(cls, *args, **kwargs):
+        if cls not in cls._instances:
+            cls._instances[cls] = super(SingletonMeta, cls).__call__(*args, **kwargs)
+        return cls._instances[cls]
```

Comment:
While I understand what you're doing/going for here, we should document it for those less versed in py fuckery